waku/node/v2/waku_types.nim

```diff
@@ -107,7 +108,7 @@ type
     rng*: ref BrHmacDrbgContext
     switch*: Switch
     peers*: seq[FilterPeer]
-    subscribers*: seq[Subscriber]
+    subscribers*: Table[PeerID, Subscriber]
```

Comment:
This seems weird, `Subscriber` contains `PeerInfo` which has the `PeerID`. If you do this then a `Peer` can also not subscrib to multipl filters. I think this will cause various weird side effects.

Reply:
Mmm. Yeah, you're right. Indexing on PeerID (which has a fast hash) was meant to simplify lookup when modifying the filter, but does limit the number of filters per subscriber to one. Rather than complicating this further, I'll rather revert to a Subscriber sequence and just filter on the PeerID already included in the Subscriber.

Reply:
Reverted to seq implementation in 3de5025